### ISSUE TITLE
Java packages for ApplicationProvider must be consistent Scala version

### ISSUE THREAD

**Issue by marcospereira** [playframework/playframework#7446]
We usually have `play.api.something` for Scala and `play.something` for Java. This is not consistent for `ApplicationProvider`:

1. Java version is `play.server.ApplicationProvider`
2. Scala version is `play.core.ApplicationProvider`

See https://github.com/playframework/playframework/pull/7300#pullrequestreview-36584049.

**Comment by mkurz** [comment]
> ... We should add an issue for 2.7.x though.

It seems there will be 2.7 before 3.0. Is it save to add a GitHub milestone for 2.7 already?

**Comment by richdougherty** [comment]
Is `ApplicationProvider` is a part of Play's API for users? I don't think users should ever reference an `ApplicationProvider` in their code. It's more of a bridging type between the server and the user `Application`. If it's not part of the public API then perhaps we have more flexibility about where we put it.

@mkurz, I see there's a milestone for _2.7.0_ available now.

**Comment by fmeriaux** [comment]
I approve.

See also : https://github.com/playframework/playframework/issues/5816

**Comment by marcospereira** [comment]
@fmeriaux no. #5816 is about an overall change while this is a very small and specific change.
